src/main/java/org/stellar/sdk/requests/ResponseHandler.java

```diff
@@ -28,7 +28,11 @@ public T handleResponse(final Response response) throws IOException, TooManyRequ
     try {
       // Too Many Requests
       if (response.code() == 429) {
-        int retryAfter = Integer.parseInt(response.header("Retry-After"));
+        
+        int retryAfter = -1;
```

Comment:
@tamirms @stellar/anchor-eng Do you have an opinion on what the default value we should be able to use here and potential impact on applications that may be consuming it?

Reply:
My wondering is that because this value is always positive, folks may be doing things like using the value as a duration in addition with current time to find a new time, or calling Time.sleep with the value, or any other number of things. Throwing a negative value into the mix may have unintended consequences and it may be safer to make the value an `Optional`.

Reply:
I agree that returning an optional would be the best

Reply:
I think the notion of a server provided retry interval introduces un-necessary coupling between server and client and a pattern to avoid, this could be the opportunity to remove `retryAfter` from `TooManyRequestsException` as a breaking api change to indicate clearer expectations to clients to determine their own error handling strategy(back-off algo, etc).

Reply:
The `Retry-After` header is a standard header to use to communicate how long a client needs to backoff for. Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

The SDK is just providing access to that header, which seems reasonable. Developers can still determine their own error handling strategy.

Reply:
IMO, making it optional is the best solution, as developer can decide default retry value themself

Reply:
agreed, changing  to `Optional<Integer>` is good trade-off for handling retyAfter, it's worth introducing the breaking change that comes with it on  `TooManyRequestsException.getRetryAfter()` as one that will alert clients that were using it, to leverage the extra info provided in notPresent to decide how to handle.

Reply:
I agree the ideal interface is an Optional, however a call needs to be made on whether breaking the API of the SDK is worth it. If not, which I suspect would be the case, using a reasonable default would be a great way to move forward with minimal impact.

Presumedly Horizon has a default/recommended value and we could use that same value here. They don't need to be the same, and it doesn't really matter if they change, as this is just a failsafe.

cc @mollykarcher I suspect you or your team can make a call if the Java SDK is accepting breaking API changes at the moment.

Reply:
I'm most concerned here that `horizon.stellar.org` isn't populating this header, which we'll look into and track separately in the #go monorepo. But as to whether we are accepting breaking changes to this SDK, _technically_ this SDK is pre-1.0, so professes no backwards compatibility guarantees. I definitely think this is less than ideal, which is why I want to cut 1.0 soon (see https://github.com/stellar/java-stellar-sdk/issues/469) to cement expectations with developers. But given the current state of things, I think going with the breaking change here is acceptable.